### PR TITLE
Add neovim profile

### PR DESCRIPTION
**Pull request by Tus1688**
I have been sketchy about neovim extensions security, as there are couple extensions that required user to run bash script after pulling some code from git repository. 
My doubts are: 
1. `disable-xdg` is not a problem for me 
2. I am not sure to add it to `firecfg.config` as vim.profile wasn't there
3. There are many plugins manager with various installation path

please let me know if I miss something :)

### CODE REVIEW

File: etc/profile-m-z/nvim.profile [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Firejail profile for neovim                                                                                                                                                                                           
+# Description: Nvim is open source and freely distributable
+# This file is overwritten after every install/update                                                                                                                                                                
+# Persistent local customizations                                                                                                                                                                                    
+include nvim.local                                                                                                                                                                                                    
+# Persistent global definitions                                                                                                                                                                                      
+include globals.local 
+
+noblacklist ${HOME}/.vim
```
Comment (rusty-snake) [inline etc/profile-m-z/nvim.profile:9]: nvim also uses
- `.cache/nvim`
- `.local/share/nvim`

File: etc/profile-m-z/nvim.profile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+include disable-programs.inc                                                                                                                                                                                         
+include disable-devel.inc                                                                                                                                                                                            
+include disable-passwdmgr.inc                                                                                                                                                                                        
+include disable-xdg.inc                                                                                                                                                                                              
+include disable-write-mnt.inc                                                                                                                                                                                        
+include whitelist-runuser-common.inc
```
Comment (rusty-snake) [inline etc/profile-m-z/nvim.profile:17]:
```suggestion

include whitelist-runuser-common.inc                                                                                                                                                                                 
```

File: etc/profile-m-z/nvim.profile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+include disable-common.inc                                                                                                                                                                                           
+include disable-programs.inc                                                                                                                                                                                         
+include disable-devel.inc                                                                                                                                                                                            
+include disable-passwdmgr.inc                                                                                                                                                                                        
+include disable-xdg.inc                                                                                                                                                                                              
+include disable-write-mnt.inc
```
Comment (rusty-snake) [inline etc/profile-m-z/nvim.profile:16]:
```suggestion
                                                                                                                                                                                       
```

File: etc/profile-m-z/nvim.profile [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+include globals.local 
+
+noblacklist ${HOME}/.vim                                                                                                                                                                                             
+                                                                                                                                                                                                                     
+include disable-common.inc                                                                                                                                                                                           
+include disable-programs.inc                                                                                                                                                                                         
+include disable-devel.inc                                                                                                                                                                                            
+include disable-passwdmgr.inc
```
Comment (rusty-snake) [inline etc/profile-m-z/nvim.profile:14]:
```suggestion                                                                                                                                                                                      
```

File: etc/profile-m-z/nvim.profile [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+# Persistent global definitions                                                                                                                                                                                      
+include globals.local 
+
+noblacklist ${HOME}/.vim                                                                                                                                                                                             
+                                                                                                                                                                                                                     
+include disable-common.inc                                                                                                                                                                                           
+include disable-programs.inc                                                                                                                                                                                         
+include disable-devel.inc
```
Comment (rusty-snake) [inline etc/profile-m-z/nvim.profile:13]: sort

File: etc/profile-m-z/nvim.profile [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+include disable-passwdmgr.inc                                                                                                                                                                                        
+include disable-xdg.inc                                                                                                                                                                                              
+include disable-write-mnt.inc                                                                                                                                                                                        
+include whitelist-runuser-common.inc                                                                                                                                                                                 
+
+# Allows files commonly used by IDEs                                                                                                                                                                                 
+include allow-common-devel.inc
```
Comment (rusty-snake) [inline etc/profile-m-z/nvim.profile:20]: Useless

File: etc/profile-m-z/nvim.profile [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+# Allows files commonly used by IDEs                                                                                                                                                                                 
+include allow-common-devel.inc                                                                                                                                                                                       
+                                                                                                                                                                                                                     
+caps.drop all                                                                                                                                                                                                        
+netfilter                                                                                                                                                                                                            
+nodbus
```
Comment (rusty-snake) [inline etc/profile-m-z/nvim.profile:24]: Deprecated

File: etc/profile-m-z/nvim.profile [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+noroot                                                                                                                                                                                                               
+notv                                                                                                                                                                                                                 
+nou2f                                                                                                                                                                                                                
+novideo                                                                                                                                                                                                              
+protocol unix,inet,inet6                                                                                                                                                                                             
+seccomp
```
Comment (rusty-snake) [inline etc/profile-m-z/nvim.profile:34]: can we use
- seccomp-block-secondary
- tracelog
- shell none
- noprinters
- no3d
- machine-id
- ipc-namespace
- net none
- disable-X11
- disable wayland
- disable runuser

**Comment by rusty-snake** [comment]
> I am not sure to add it to firecfg.config as vim.profile wasn't there

IMHO we shouldn't add it.

**Comment by Tus1688** [comment]
can we use

- seccomp-block-secondary
- tracelog
- shell none
- noprinters
- no3d
- machine-id
- ipc-namespace
- net none
- disable-X11
- disable wayland
- disable runuser

Sure, I will let you know as soon as I get back to my machine.

**Comment by Tus1688** [comment]
I have sort and do trial and error. Is there anything I could provide to you?
my questions & doubts are:
1. Why removing `disable-passwdmgr.inc` ?
2. `net none` make neovim plugin manager is not working as it supposed to be. Are we trying to mimick vs code trusted workspace feature? If so, maybe adding `read-only` on `${HOME}/.vim` could make it perfect.
3. I add `nonewprivs`
4. I couldn't test `noprinters`

File: etc/profile-m-z/nvim.profile [hunk 9]
```diff
@@ -1,39 +1,46 @@
...
+nodvd
+nogroups
+noinput
+nonewprivs
+noroot
+notv
+nou2f
+novideo
+protocol unix,inet,inet6
+seccomp.block-secondary
```
Comment (rusty-snake) [inline etc/profile-m-z/nvim.profile:38]: Why was `seccomp` removed?

File: etc/profile-m-z/nvim.profile [hunk 10]
```diff
@@ -1,39 +1,46 @@
...
+
+include whitelist-runuser-common.inc
+
+ipc-namespace
+machine-id
+net none
+netfilter
+no3d
+dbus-user none
+dbus-system none
```
Comment (rusty-snake) [inline etc/profile-m-z/nvim.profile:28]: Move down as per template.

File: etc/profile-m-z/nvim.profile [hunk 11]
```diff
@@ -1,39 +1,46 @@
...
+
+blacklist ${RUNUSER}/wayland-*
+blacklist ${RUNUSER}
+
+include whitelist-runuser-common.inc
+
+ipc-namespace
+machine-id
+net none
+netfilter
```
Comment (rusty-snake) [inline etc/profile-m-z/nvim.profile:25]:
```suggestion
```

File: etc/profile-m-z/nvim.profile [hunk 12]
```diff
@@ -1,39 +1,46 @@
...
+noblacklist ${HOME}/.vim
+noblacklist ${HOME}/.cache/nvim
+noblacklist ${HOME}/.local/share/nvim
+
+include disable-common.inc
+include disable-devel.inc
+include disable-programs.inc
+include disable-xdg.inc
+
+blacklist ${RUNUSER}/wayland-*
```
Comment (rusty-snake) [inline etc/profile-m-z/nvim.profile:17]:
```suggestion
```

File: etc/profile-m-z/nvim.profile [hunk 13]
```diff
@@ -1,39 +1,46 @@
...
-include disable-devel.inc                                                                                                                                                                                            
-include disable-passwdmgr.inc                                                                                                                                                                                        
-include disable-xdg.inc                                                                                                                                                                                              
-include disable-write-mnt.inc                                                                                                                                                                                        
-include whitelist-runuser-common.inc                                                                                                                                                                                 
+noblacklist ${HOME}/.vim
+noblacklist ${HOME}/.cache/nvim
+noblacklist ${HOME}/.local/share/nvim
```
Comment (rusty-snake) [inline etc/profile-m-z/nvim.profile:10]: Add them to disable-programs.inc.

File: etc/profile-m-z/nvim.profile [hunk 14]
```diff
@@ -1,39 +1,46 @@
-# Firejail profile for neovim                                                                                                                                                                                           
+# Firejail profile for neovim
 # Description: Nvim is open source and freely distributable
-# This file is overwritten after every install/update
```
Comment (rusty-snake) [inline etc/profile-m-z/nvim.profile]: Should be kept.

**Comment by rusty-snake** [comment]
1. It no longer exists.

File: etc/profile-m-z/nvim.profile (hunk 9 above)
Comment (Tus1688) [reply etc/profile-m-z/nvim.profile:38]: My bad, I thought it is same

**Comment by Tus1688** [comment]
done :)

File: etc/inc/disable-programs.inc [hunk 15]
```diff
@@ -939,6 +940,7 @@ blacklist ${HOME}/.local/share/newsboat
 blacklist ${HOME}/.local/share/nheko
 blacklist ${HOME}/.local/share/nomacs
 blacklist ${HOME}/.local/share/notes
+blacklist ${HOME}/.local/share/nvim
 blacklist ${HOME}/.local/share/ocenaudio
 blacklist ${HOME}/.local/share/okular
 blacklist ${HOME}/.local/share/onlyoffice
```
Comment (kmk3) [inline etc/inc/disable-programs.inc:946]: Add these to disable-common.inc (near each equivalent vim entry):

```firejail
blacklist-nolog ${HOME}/.local/share/nvim

read-only ${HOME}/.config/nvim
read-only ${HOME}/.local/share/nvim
```

**Review by kmk3 — COMMENTED** [review]
Config dir and dc entries are missing.

**Comment by Tus1688** [comment]
Thank you for code review, is there anything I could do for you?

**Comment by rusty-snake** [comment]
> Where did that one come from?

cli/tui program, no gui.

**Comment by kmk3** [comment]
@rusty-snake commented [on Jan 13](https://github.com/netblue30/firejail/pull/4841#issuecomment-1011912265):

> > Where did that one come from?
> 
> cli/tui program, no gui.

What I meant is that it does not exist on master.

**Comment by rusty-snake** [comment]
https://github.com/netblue30/firejail/blob/cd0482c0a48d5fdc7b3878adff42b76643549b73/etc/templates/profile.template#L105-L106

**Comment by kmk3** [comment]
@rusty-snake commented [on Jan 13](https://github.com/netblue30/firejail/pull/4841#issuecomment-1012432146):

> https://github.com/netblue30/firejail/blob/cd0482c0a48d5fdc7b3878adff42b76643549b73/etc/templates/profile.template#L105-L106

Ah I see, nevermind then.

I saw this:

> - disable-X11
> - disable wayland

And since "disable-X11.inc" exists, I thought you meant "disable-wayland.inc".

**Comment by Tus1688** [comment]
> I think it would be nice to have at least two different configurations:
> 
> "dev mode": Enables all the plugins, but can't edit vim/neovim config files
> "admin mode": Does not enable any plugin, but can edit vim/neovim config
> files
> The second one can be achieved by running nvim --noplugin outside of
> firejail.
> 
> The first one could be done by leaving the config files read-only in this
> profile.
> 
> Thoughts?

I am not security expert. But, I have been wondering how to install neovim plugin outside package manager? 

> Then if (neo)vim is ever opened outside of firejail (which would be needed to
> edit files in e.g.: ~/.config/firejail), it's game over

.

**Comment by kmk3** [comment]
@Tus1688 commented [on Jan 14](https://github.com/netblue30/firejail/pull/4841#issuecomment-1012682254):

> > I think it would be nice to have at least two different configurations:
> > 
> > "dev mode": Enables all the plugins, but can't edit vim/neovim config files
> > "admin mode": Does not enable any plugin, but can edit vim/neovim config
> > files
> > 
> > The second one can be achieved by running nvim --noplugin outside of
> > firejail.
> > 
> > The first one could be done by leaving the config files read-only in this
> > profile.
> > 
> > Thoughts?
> 
> I am not a security expert. But, I have been wondering how to install neovim
> plugin outside package manager?

I'm not sure what you mean, but there are 3 ways to install plugins that I know
of:

1. The system package manager

For example, on Arch/Artix there are packages for vim plugins, some/all of
which are part of the "vim-plugins" group:

<details>
<summary>vim-plugins</summary>
<p>

```console
$ pacman -Sgq vim-plugins
vim-airline
vim-airline-themes
vim-ale
vim-align
vim-ansible
vim-bufexplorer
vim-coverage-highlight
vim-csound
vim-ctrlp
vim-easymotion
vim-editorconfig
vim-fugitive
vim-gitgutter
vim-grammalecte
vim-indent-object
vim-jad
vim-jedi
vim-latexsuite
vim-molokai
vim-nerdcommenter
vim-nerdtree
vim-pastie
vim-seti
vim-supertab
vim-surround
vim-syntastic
vim-tabular
vim-tagbar
vim-ultisnips
vim-vital
```

</p>
</details>

These can be installed with `pacman -S` like any other package.

2. The built-in package manager (vim 8+/neovim)
3. A (neo)vim plugin manager, such as vim-plug, vundle or pathogen.vim

In the first case, someone from the distro has to write and maintain a recipe
to package the plugin (and ideally also read the diffs between versions), so
there is at least a "buffer"/some level of quality control in the way of an
attack, which means that there is a higher likelihood of it being spotted
before making its way to end users (just like with distro packages in general).

AFAIK in the other cases, installing a plugin is equivalent to cloning a given
plugin repo (such as directly from github.com) into the (neo)vim plugins
directory (so not too different from the "curl into bash" approach).  In this
case, probably all it takes is one malicious commit (and maybe also a new
version tag) followed by a user running a command to update the installed
plugins.

See also:

* `:help plugin`
* <https://wiki.archlinux.org/title/Vim#Plugins>

**Comment by Tus1688** [comment]
I am sorry if there is a language barrier between us, as English is not my native language. So basically, in current neovim profile it is impossible to use vim plugin manager such as vim plug, and it is quite annoying (fetching & pulling each directory) if the user doesn't have "certain" package manager to install neovim plugins.

**Comment by netblue30** [comment]
Let's grab it as is, without firecfg.config, so is not sandboxed automatically. New development on new commits/prs. Thanks @Tus1688

**Pull request merged**